src/analyzer.rs

```diff
@@ -290,19 +290,37 @@ impl<'run, 'src> Analyzer<'run, 'src> {
     Ok(())
   }
 
-  fn resolve_alias(
-    recipes: &Table<'src, Rc<Recipe<'src>>>,
-    alias: Alias<'src, Name<'src>>,
+  fn resolve_alias<'a>(
+    modules: &'a Table<'src, Justfile<'src>>,
+    recipes: &'a Table<'src, Rc<Recipe<'src>>>,
+    alias: Alias<'src, Namepath<'src>>,
   ) -> CompileResult<'src, Alias<'src>> {
     // Make sure the target recipe exists
-    match recipes.get(alias.target.lexeme()) {
-      Some(target) => Ok(alias.resolve(Rc::clone(target))),
+    match Self::recipe_from_path(alias.target.as_ref(), modules, recipes) {
+      Some(target) => Ok(alias.resolve(target)),
       None => Err(alias.name.token.error(UnknownAliasTarget {
         alias: alias.name.lexeme(),
-        target: alias.target.lexeme(),
+        target: alias.target,
       })),
     }
   }
+
+  fn recipe_from_path<'a>(
+    path: &[Name],
+    mut modules: &'a Table<'src, Justfile<'src>>,
+    mut recipes: &'a Table<'src, Rc<Recipe<'src>>>,
+  ) -> Option<Rc<Recipe<'src>>> {
+    let name = path.last()?;
```

Comment:
What happens if `.last()` return `None`? This will be treated as an unknown alias target, but is in fact a bug, since namepath should be non-empty, and thus if `.last()` returns `None` it should be an internal error. This is why we shouldn't iplement `Deref` for `Namepath`, because the function signature `fn last(&self) -> Option<Name>` doesn't make sense. It should be `fn last(&self) -> Name`.

Reply:
Will `expect()` panic now saying that `Namepath` can't be empty